docs/rebase.md

```diff
@@ -157,3 +157,148 @@ git commit -m "update buildfiles"
+These references are passed to `rebase.py` using `AMD64_RELEASE` and `ARM64_RELEASE` environment variables, for example:
+```
+AMD64_RELEASE=registry.ci.openshift.org/ocp/release:4.13.0-0.nightly-2023-01-27-165107 \
+ARM64_RELEASE=registry.ci.openshift.org/ocp-arm64/release-arm64:4.13.0-0.nightly-arm64-2023-01-30-010253 \
+./scripts/auto-rebase/rebase.py
+```
+
+#### Testing locally
+
+For testing `rebase.py` locally, following env vars can be useful:
+- `TOKEN` expects a GitHub Personal Access Token which can be generated [here](https://github.com/settings/tokens).
+   Use it instead of `APP_ID` and `KEY`.
+- `DRY_RUN` instructs script to not make any changes on the repo (i.e. git push, create PR, post comment, etc.) - instead actions are logged and script continues.
+- `BASE_BRANCH` forces script to diff results against different branch than what was checked out when script started running
+   (useful when testing changes exist on local branch that is not `main` - otherwise, script would want to create a PR with base being branch does not exists on remote).
+
+```bash
+TOKEN=ghp_... \
+ORG=openshift \
+DRY_RUN=y \
+REPO=microshift \
+AMD64_RELEASE=registry.ci.openshift.org/ocp/release:4.13.0-0.nightly-2023-01-27-165107 \
+ARM64_RELEASE=registry.ci.openshift.org/ocp-arm64/release-arm64:4.13.0-0.nightly-arm64-2023-01-30-010253 \
+./scripts/auto-rebase/rebase.py
+```
+
+Script also can be ran against a private fork in non-dry run which is helpful to verify that the communication with remote repository is as expected and so is created PR.
+In such case `ORG` env var needs to be set to GitHub username:
+
+```bash
+TOKEN=ghp_... \
+ORG=USER_NAME \
+REPO=microshift \
+AMD64_RELEASE=registry.ci.openshift.org/ocp/release:4.13.0-0.nightly-2023-01-27-165107 \
+ARM64_RELEASE=registry.ci.openshift.org/ocp-arm64/release-arm64:4.13.0-0.nightly-arm64-2023-01-30-010253 \
+./scripts/auto-rebase/rebase.py
+```
+
+#### Testing in CI
+
+> Note: Rehearsing Rebase Prow Job without dry run can result in creation of rebase PR in openshift/microshift.
+
+To test changes in context of "production" (CI Prow Job) environment it's recommended to:
+- Create a dummy PR in `release` repo for rehearsing the rebase job that checks out changes under test
+  - Example of `ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-commands.sh` from [PR](https://github.com/openshift/release/pull/35875/files)
+    ```
+    #!/bin/bash
+
+    git remote add pmtk https://github.com/pmtk/microshift.git
+    git fetch pmtk
+    git switch --track pmtk/csi-rebase-script
+
+    # export DRY_RUN=y
+    
+    ./scripts/auto-rebase/rebase_job_entrypoint.sh
+    
+    git diff csi-rebase-script
```

Comment:
Can we update the script to get the user name from the command line, or at least, to set it in a var in one place?

Reply:
This snippet shows example modifications to https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-commands.sh which isn't intended to be executed locally (even more so in interactive environment), but added possibility to override ORG and REPO (rebase_job_entrypoint.sh) and reworded this section